utils/mil_tools/nodes/label_me_bag.py

```diff
@@ -0,0 +1,268 @@
+      else:
+          self._print("\t\t{}/{} images labeled in this bag ({:.1%})", total_xml_count, total_img_count, total_xml_count/total_img_count)
+      return total_xml_count, total_img_count
+
+  def _parse_label_xml(self, filename):
+      msg = LabeledObjects()
+      e = xml.etree.ElementTree.parse(filename).getroot()
+      for obj in e.findall('object'):
+          obj_msg = LabeledObject()
+          obj_msg.name = obj.find('name').text
+          obj_msg.id = int(obj.find('id').text)
+          parent = obj.find('parts').find('ispartof').text
+          if parent != None:
+            obj_msg.parent_id = int(parent)
+          attributes = obj.find('attributes').text
+          if attributes != None:
+              obj_msg.attributes = attributes
+          polygon = obj.find('polygon')
+          for pt in polygon.findall('pt'):
+              x = int(pt.find('x').text)
+              y = int(pt.find('y').text)
+              obj_msg.polygon.append(Point(x, y, 0))
+          msg.objects.append(obj_msg)
+      return msg
+
+  def _get_labels(self, bag_config):
+      labels = {}
+      for segment in bag_config['segments']:
+          labels[segment['topic']] = {}
+      for segment in bag_config['segments']:
+          path = os.path.join(self.labelme_dir, 'Annotations', segment['name'], segment['topic'].replace('/', '@'))
+          self._print("\t\tExtracting LabelMe annotations from {}", path)
+          if not os.path.isdir(path):
+              continue
+          for xmlfile in os.listdir(path):
+              stamp = xmlfile.split('.xml', 1)
+              assert len(stamp) == 2
+              labels[segment['topic']][stamp[0]] = os.path.join(path, xmlfile)
+      return labels
+
+  def _extract_labels_bag(self, bag_config, out):
+      label_files = self._get_labels(bag_config)
+      bag = rosbag.Bag(bag_config['file'])
+      out = rosbag.Bag(out, mode='w')
+      _, _, first_time = bag.read_messages().next()
+      for topic, msg, t in bag.read_messages():
+        if msg._type == 'sensor_msgs/Image':
+            if topic in label_files:
+                if str(msg.header.stamp) in label_files[topic]:
+                    labels = self._parse_label_xml(label_files[topic][str(msg.header.stamp)])
+                    labels.header = msg.header
+                    out.write(topic+'/labels', labels, t)
+        out.write(topic, msg, t)
+      out.flush()
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Generates rosbags based on LabelMe data and visa/versa')
+    parser.add_argument('config', type=str,
+                        help='YAML file specifying what bags to read and extract images from. See example YAML for details')
+    parser.add_argument('--labelme-dir', '-d', dest='dir',type=str, required=True,
+                        help='root directory of labelme instalation')
+    parser.add_argument('--extract-labels', '-e', dest='extract_labels', action='store_true',
```

Comment:
I don't know if extract is the right word for this. I would associate that more with the image extraction process. Maybe use inject?